src/components/booking/BookingForm.tsx

```diff
@@ -0,0 +1,78 @@
+// BookingForm.tsx
+import React, { useState } from 'react';
+import { Box, Stack } from '@mui/material';
+import { Dayjs } from 'dayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { DatePicker } from '@mui/x-date-pickers/DatePicker';
+import ButtonPrimary from '../../components/button/ButtonPrimary';
+import { BookingFormProps } from '../../types/booking.types';
+
+const BookingForm: React.FC<BookingFormProps> = ({ onSubmit, data }) => {
+  const [selectedDates, setSelectedDates] = useState<[Dayjs | null, Dayjs | null]>([null, null]);
+
+  const { availableDates, accommodationId } = data || {};
+
+  const shouldDisableDate = (date: Dayjs) => {
+    const dateString = date.format('YYYY-MM-DD');
+    return !availableDates?.some((range) => dateString >= range[0] && dateString <= range[1]);
```

Comment:
return !availableDates?.some(([start, end]) => dateString >= start && dateString <= end);
make understandable names your code will be much clearer